dockerfiles/go-1.22.Dockerfile

```
@@ -1,14 +1,16 @@
 
-COPY go.mod /app/go.mod
-COPY go.sum /app/go.sum
+# Ensures the container is re-built if go.mod or go.sum changes
+ENV CODECRAFTERS_DEPENDENCY_FILE_PATHS="go.mod,go.sum"
 
 WORKDIR /app
 
-# Starting from Go 1.20, the go standard library is no loger compiled
-# setting the GODEBUG environment to "installgoroot=all" restores the old behavior
+# .git & README.md are unique per-repository. We ignore them on first copy to prevent cache misses
+COPY --exclude=.git --exclude=README.md . /app
```

Hadolint (2.12.0)

[error] invalid flag: --exclude

(DL1000)